### PR TITLE
Get projects after workspace has been already initialized

### PR DESCRIPTION
**Pull request by vzhukovs**
This commit also includes API reorganization. Move client service implementations from the core-api to the core-app. Code cleanup. Removed dependency to the `javax.inject` from the core-api to avoid duplicate using `@Inject` annotation from the `com.google.inject` and `javax.inject`. (Client side uses only `com.google.inject.Inject` annotation)

Created two components:
**WsAgentInitializer** - which is responsible to initialize workspace component on the client side. Component consumes developer machine object at startup stage.
**WsAgentMessageBusProvider** - which is responsible to initialize message bus after workspace has been initialized.

API movements:
`OAuthServiceClient` was moved from the `org.eclipse.che.ide.api.auth` to the `org.eclipse.ide.api.oauth`

Related issue: https://github.com/eclipse/che/issues/2629

@vparfonov @ashumilova @azatsarynnyy review it, please.

### CODE REVIEW

File: ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/machine/WsAgentInitializer.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+/**
+ * Initialize client's workspace agent component with specific developer machine. Usually, this interface
+ * is not intended to be implemented by the third-party components.
+ *
+ * @author Vlad Zhukovskyi
+ * @see DevMachine
+ * @since 5.0.0
+ */
+public interface WsAgentInitializer {
```
Comment (azatsarynnyy) [inline ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/machine/WsAgentInitializer.java]: As far as I understood this interface is used in IDE app only and not intended to be used by extensions.
So, if it isn't a part of IDE public API why do we need it in IDE API? May be it would be better to move it in IDE app module?

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/machine/WsAgentStateController.java [hunk 2]
```diff
@@ -139,7 +150,7 @@ public WsAgentState getState() {
         return state;
     }
 
-    /** Returns instance of {@link MachineMessageBus}. */
+    /** {@inheritDoc} */
     public Promise<MessageBus> getMessageBus() {
```
Comment (azatsarynnyy) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/machine/WsAgentStateController.java:155]: _@Override_ annotation is missed

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/machine/WsAgentStateController.java [hunk 3]
```diff
@@ -81,23 +93,22 @@ public WsAgentStateController(WorkspaceServiceClient workspaceServiceClient,
...
     }
 
-    /**
-     * Initialize state of the agent.
-     *
-     * @param devMachine
-     *         development machine instance
-     */
+    /** {@inheritDoc} */
     public void initialize(DevMachine devMachine) {
```
Comment (azatsarynnyy) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/machine/WsAgentStateController.java]: _@Override_ annotation is missed

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/ssh/SshServiceClientImpl.java [hunk 4]
```diff
@@ -29,6 +31,7 @@
  *
  * @author Sergii Leschenko
  */
+@Singleton
```
Comment (azatsarynnyy) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/ssh/SshServiceClientImpl.java]: Why did you add this annotation?
It's already bound as singleton in CoreGinModule.

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/machine/WsAgentStateController.java [hunk 5]
```diff
@@ -81,23 +93,22 @@ public WsAgentStateController(WorkspaceServiceClient workspaceServiceClient,
...
         this.availableServices = new ArrayList<>();
     }
 
-    /**
-     * Initialize state of the agent.
-     *
-     * @param devMachine
-     *         development machine instance
-     */
+    /** {@inheritDoc} */
```
Comment (azatsarynnyy) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/machine/WsAgentStateController.java]: What is the purpose of an empty javadoc which contains _@inheritDoc_ tag only?
Javadoc tool and all popular IDEs override it automatically when generating documentation.
IMO we should add it only in case of 'real' overriding javadoc like [here](https://github.com/eclipse/che/blob/master/ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/wizard/AbstractWizardPage.java#L34).
WDYT?

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/oauth/OAuthServiceClientImpl.java [hunk 6]
```diff
@@ -8,18 +8,21 @@
...
 import org.eclipse.che.ide.rest.AsyncRequestCallback;
 import org.eclipse.che.ide.rest.AsyncRequestFactory;
 import org.eclipse.che.ide.rest.RestContext;
 
-import javax.inject.Inject;
-
 /**
  * @author Sergii Leschenko
  */
+@Singleton
```
Comment (azatsarynnyy) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/oauth/OAuthServiceClientImpl.java]: Why did you add this annotation?
It's already bound as singleton in CoreGinModule.

**Review by azatsarynnyy — COMMENTED** [review]
:+1:

File: ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/machine/WsAgentInitializer.java (hunk 1 above)
Comment (vzhukovs) [reply ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/machine/WsAgentInitializer.java]: Yes, you're right, I'll move it to the IDE app

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/machine/WsAgentStateController.java (hunk 5 above)
Comment (vzhukovs) [reply ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/machine/WsAgentStateController.java]: It's just personal habbit. I'll remove it.

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/machine/WsAgentStateController.java (hunk 3 above)
Comment (vzhukovs) [reply ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/machine/WsAgentStateController.java]: Fixed

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/machine/WsAgentStateController.java (hunk 2 above)
Comment (vzhukovs) [reply ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/machine/WsAgentStateController.java:155]: Fixed

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/ssh/SshServiceClientImpl.java (hunk 4 above)
Comment (vzhukovs) [reply ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/ssh/SshServiceClientImpl.java]: Yes, sure, fixed

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/oauth/OAuthServiceClientImpl.java (hunk 6 above)
Comment (vzhukovs) [reply ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/oauth/OAuthServiceClientImpl.java]: Also fixed

**Review by ashumilova — APPROVED** [review]

**Review by azatsarynnyy — APPROVED** [review]
